### PR TITLE
Added implementations of CRUD methods to UserRepository. Added tests …

### PR DESCRIPTION
**Pull request by SergeyPrudnikov-dev**
…for UserRepository.

### CODE REVIEW

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    
+    /**
+     * Finds and returns an object with the given identifier.
+     *
+     * @param aLong of the entity.
+     * @return T - entity.
+     */
+    @Override
+    public User read(Long aLong) {
+        SqlSession sqlSession = MyBatisUtil.getsQLSessionFactory().openSession();
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:44]: try-with-resources?

File: src/main/java/com/epam/university/commission/util/MyBatisUtil.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * This is the class for getting the SqlSessionFactory instance.
+ * <p/>
+ * University-commission 2021  epam.com
+ * <p/>
+ * Date: 05/15/2021
+ *
+ * @author Siarhei Prudnikau1
+ */
+public class MyBatisUtil {
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/util/MyBatisUtil.java:20]: should be final

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    </select>
+
+    <update id='updateUser' parameterType='User'>
+        UPDATE "user"
+        SET
+        login = #{login},
+        password = #{password},
+        email = #{email},
+        role_id = #{role}
+        WHERE id = #{id}
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:37]: 4 space for fields

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        VALUES(#{login}, #{password}, #{email}, #{role})
+    </insert>
+
+    <select id='getUserById' parameterType='long' resultType='User'>
+        SELECT
+        id as id,
+        login as login,
+        password,
+        email as email,
+        role_id as role
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:25]: 4 space

File: src/main/resources/liquibase/db/changelog/v-1.1/db.changelog-1.1-create-user-table.xml [hunk 5]
```diff
@@ -43,4 +44,12 @@
                 startWith="1"
                 tableName="user"/>
     </changeSet>
+
+    <!--<changeSet id="2021-05-13-00" author="Siarhei Prudnikau (siarhei_prudnikau1@epam.com)">-->
```
Comment (andrei-kuzniatsou) [inline src/main/resources/liquibase/db/changelog/v-1.1/db.changelog-1.1-create-user-table.xml:48]: it looks redundant

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+/**
+ * Test to verify the functionality of the {@link UserRepository}.
+ * <p/>
+ * University-commission 2021  epam.com
+ * <p/>
+ * Date: 05/17/2021
+ *
+ * @author Siarhei Prudnikau1
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:27]: should not rely on order

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    @AfterAll
+    public void teardown() {
+        if (testUserRepository.read(userId) != null) {
+            testUserRepository.delete(userId);
+        }
+        testUserRepository = null;
+    }
+
+    @Test
+    @Order(1)
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:60]: should not be ordered

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        assertEquals(user.getRole(), newUser.getRole());
+        assertEquals(user.getId(), newUser.getId());
+        // assertEquals(user,newUser);
+        // КАК ЛУЧШЕ, ПРОВЕРИТЬ ВСЕ ПОЛЯ ИЛИ СРАЗУ ОБЪЕКТЫ?
+    }
+
+    @Test
+    @Order(2)
+    void read() {
+        if (userId == null) {
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:79]: why do we need this check?

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    private UserRepository testUserRepository;
+
+    @BeforeAll
+    void setUp() {
+        testUserRepository = new UserRepository();
+        user = new User();
+        user.setLogin(USER_LOGIN);
+        user.setPassword(USER_PASSWORD);
+        user.setEmail(USER_EMAIL);
+        user.setRole(USER_ROLE);
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:48]: create private method with params to build user and user for tests

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+ * <p/>
+ * University-commission 2021  epam.com
+ * <p/>
+ * Date: 05/17/2021
+ *
+ * @author Siarhei Prudnikau1
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class UserRepositoryTest {
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:29]: I suggest naming UserRepositoryIntegrationTest, see the difference between Unit and IntegrationTest,

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    public void teardown() {
+        if (testUserRepository.read(userId) != null) {
+            testUserRepository.delete(userId);
+        }
+        testUserRepository = null;
+    }
+
+    @Test
+    @Order(1)
+    void create() {
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:61]: name should be informative. If we follow your way, in the test report can be a lot of test with the same name

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+        newUser = testUserRepository.read(user.getId());
+        assertNotNull(newUser);
+        assertEquals(user.getLogin(), newUser.getLogin());
+        assertEquals(user.getPassword(), newUser.getPassword());
+        assertEquals(user.getEmail(), newUser.getEmail());
+        assertEquals(user.getRole(), newUser.getRole());
+        assertEquals(user.getId(), newUser.getId());
+        // assertEquals(user,newUser);
+        // КАК ЛУЧШЕ, ПРОВЕРИТЬ ВСЕ ПОЛЯ ИЛИ СРАЗУ ОБЪЕКТЫ?
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:73]: depend on the test, in this case I would like to have assertEquals(user,newUser), as fas as we have hashcode and equals

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 8 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java:79]: This check is needed if we run a separate test method, without the rest of the tests.

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+            sqlSession.commit();
+            return entity.getId();
+        }
+    }
+
+    /**
+     * Finds and returns an object with the given identifier.
+     *
+     * @param aLong of the entity.
+     * @return T - entity.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:40]: return User, param userId

File: src/main/java/com/epam/university/commission/domain/Role.java [hunk 14]
```diff
@@ -33,6 +35,21 @@ private Role(String name) {
         this.name = name;
     }
 
+    /**
+     * Returns the role by identifier.
+     *
+     * @param id is ordinal number.
+     * @return Role is role of user.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/domain/Role.java:42]: return Role enumeration

File: src/main/java/com/epam/university/commission/domain/Role.java [hunk 15]
```diff
@@ -33,6 +35,21 @@ private Role(String name) {
...
+     */
+    public static Role lookupById(int id) {
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/domain/Role.java:44]: Do we need lookup method?

File: src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
+package com.epam.university.commission.handler;
+
+import com.epam.university.commission.domain.Role;
+import org.apache.ibatis.type.EnumOrdinalTypeHandler;
+
+/**
+ * Helper class for correct type Role definition in the database.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java:7]: TypeHandler for Role enumeration

File: src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+ * University-commission 2021  epam.com
+ * <p/>
+ * Date: 05/17/2021
+ *
+ * @author Siarhei Prudnikau1
+ */
+public class RoleTypeHandler extends EnumOrdinalTypeHandler<Role> {
+
+    /**
+     * @param type for EnumOrdinalTypeHandler<>.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java:18]: Role enumeration

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 18]
```diff
@@ -0,0 +1,44 @@
+package com.epam.university.commission.mappers;
+
+import com.epam.university.commission.domain.User;
+
+/**
+ * This is the interface for mybatis to work.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:6]: to deal with User

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+ * <p/>
+ * University-commission 2021  epam.com
+ * <p/>
+ * Date: 05/15/2021
+ *
+ * @author Siarhei Prudnikau1
+ */
+public interface IUserMapper {
+    /**
+     * Insertes a given entity.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:16]: Inserts User to the database

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+     *
+     * @param user - the entity that is being inserted into the database.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:18]: {@link User}

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+     * @return id user.
+     */
+    Long insertUser(User user);
+
+    /**
+     * Finds and returns an object with the given identifier.
+     *
+     * @param id of the entity.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:26]: user id

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Insertes a given entity.
+     *
+     * @param user - the entity that is being inserted into the database.
+     * @return id user.
+     */
+    Long insertUser(User user);
+
+    /**
+     * Finds and returns an object with the given identifier.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:24]: add user to doc

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java [hunk 23]
```diff
@@ -0,0 +1,77 @@
+package com.epam.university.commission.repository.impl;
+
+import com.epam.university.commission.domain.User;
+import com.epam.university.commission.mappers.IUserMapper;
+import com.epam.university.commission.repository.api.IUserRepository;
+import com.epam.university.commission.util.MyBatisUtil;
+import org.apache.ibatis.session.SqlSession;
+
+/**
+ * Class for working the User entity with the database.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:10]: Repository to operate with User

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public User read(Long aLong) {
+        try (SqlSession sqlSession = MyBatisUtil.openSession()) {
+            IUserMapper iUserMapper = sqlSession.getMapper(IUserMapper.class);
+            return iUserMapper.getUserById(aLong);
+        }
+    }
+
+    /**
+     * Updates a given entity.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:51]: user

File: src/main/resources/db/mybatis/mybatis-config.xml [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+                <property name = "url" value = "${databaseUrlMain}"/>
+                <property name = "username" value = "${databaseUsernameMain}"/>
+                <property name = "password" value = "${databasePasswordMain}"/>
+            </dataSource>
+        </environment>
+    </environments>
+
+    <mappers>
+        <mapper resource = "db/mybatis/UserMapper.xml"/>
+        <mapper resource = "db/mybatis/FacultyMapper.xml"/>
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/mybatis-config.xml:31]: FacultyMapper?

File: gradle.properties [hunk 26]
```diff
@@ -6,3 +6,4 @@ databaseUrlMain = jdbc:postgresql://localhost:5432/uc
 databaseUsernameMain = admin_uc
 databasePasswordMain = 12345
 
+databaseDriverClassName = org.postgresql.Driver
```
Comment (andrei-kuzniatsou) [inline gradle.properties:9]: do you need this property?

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+    private User user;
+    private User newUser;
+
+    @BeforeAll
+    void setUp1() {
+        sTestUserRepository = new UserRepository();
+    }
+
+    @BeforeEach
+    void setUp2() {
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:44]: you don't have to create 2 setup methods

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class UserRepositoryIntegrationTest {
+
+    private static final String USER_EMAIL = "test_email@gmail.com";
+    private static final String USER_LOGIN = "test_login";
+    private static final String USER_PASSWORD = "test_password";
+    private static final Role USER_ROLE = Role.ADMIN;
+
+    private static UserRepository sTestUserRepository;
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:33]: Why static?
name -> userRepository

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+ *
+ * @author Siarhei Prudnikau1
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class UserRepositoryIntegrationTest {
+
+    private static final String USER_EMAIL = "test_email@gmail.com";
+    private static final String USER_LOGIN = "test_login";
+    private static final String USER_PASSWORD = "test_password";
+    private static final Role USER_ROLE = Role.ADMIN;
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:31]: How many time it is used id the code?

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java [hunk 30]
```diff
@@ -0,0 +1,106 @@
...
+
+    private static UserRepository sTestUserRepository;
+    private Long userId;
+    private User user;
+    private User newUser;
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:36]: why is it global?

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java [hunk 31]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Test
+    void testCreateUser() {
+        userId = sTestUserRepository.create(user);
+        assertNotNull(userId);
+
+        newUser = sTestUserRepository.read(user.getId());
+        assertNotNull(newUser);
+        assertEquals(user, newUser);
```
Comment (andrei-kuzniatsou) [inline src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:62]: let's discuss

File: src/main/java/com/epam/university/commission/domain/Role.java (hunk 15 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/domain/Role.java:44]: We do not use this method. So we don't need this lookupById (int id) method.

File: src/main/resources/db/mybatis/mybatis-config.xml (hunk 25 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/mybatis-config.xml:31]: I started writing FacultyMapper.

File: gradle.properties (hunk 26 above)
Comment (SergeyPrudnikov-dev) [reply gradle.properties:9]: No, I wanted to combine gradle.property and mybatis.gradle but couldn't. I forgot to remove this parameter.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java (hunk 28 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:33]: The static modifier does not need in fixed version of the class.

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java (hunk 24 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:51]: Fixed.

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java (hunk 1 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/repository/impl/UserRepository.java]: Fixed.

File: src/main/java/com/epam/university/commission/util/MyBatisUtil.java (hunk 2 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/util/MyBatisUtil.java]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 3 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 4 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml]: Fixed.

File: src/main/resources/liquibase/db/changelog/v-1.1/db.changelog-1.1-create-user-table.xml (hunk 5 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/liquibase/db/changelog/v-1.1/db.changelog-1.1-create-user-table.xml]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 6 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 7 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 8 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 9 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 10 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 11 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java (hunk 12 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryTest.java]: Fixed.

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java (hunk 13 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:40]: Fixed.

File: src/main/java/com/epam/university/commission/domain/Role.java (hunk 15 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/domain/Role.java:44]: I deleted this method.

File: src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java (hunk 16 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java:7]: Fixed.

File: src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java (hunk 17 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/handler/RoleTypeHandler.java:18]: Fixed.

File: src/main/java/com/epam/university/commission/domain/Role.java (hunk 14 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/domain/Role.java:42]: deleted

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 18 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:6]: Fixed.

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 19 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:16]: Fixed.

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 20 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:18]: Fixed.

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 21 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:26]: Fixed.

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 22 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:24]: Fixed.

File: src/main/java/com/epam/university/commission/repository/impl/UserRepository.java (hunk 23 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/repository/impl/UserRepository.java:10]: Fixed.

File: gradle.properties (hunk 26 above)
Comment (SergeyPrudnikov-dev) [reply gradle.properties:9]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java (hunk 28 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:33]: Fixed.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java (hunk 29 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:31]: 1 time. We don't need static final modifiers?

File: src/main/resources/db/mybatis/mybatis-config.xml (hunk 25 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/mybatis-config.xml:31]: I will remove this line from this commit.

File: src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java (hunk 27 above)
Comment (SergeyPrudnikov-dev) [reply src/test/java/com/epam/university/commission/repository/impl/UserRepositoryIntegrationTest.java:44]: Please check out my new version of this class.

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java [hunk 32]
```diff
@@ -0,0 +1,44 @@
+package com.epam.university.commission.mappers;
+
+import com.epam.university.commission.domain.User;
+
+/**
+ * This is the interface to deal with {@link User}.
```
Comment (andrei-kuzniatsou) [inline src/main/java/com/epam/university/commission/mappers/IUserMapper.java:6]: it is mapper interface

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+<mapper namespace='com.epam.university.commission.mappers.IUserMapper'>
+    <resultMap type='User' id='UserResult'>
+        <id property='id' column='id'/>
+        <result property='login' column='login'/>
+        <result property='password' column='password'/>
+        <result property='email' column='email'/>
+        <result property='role' column='role_id'/>
+    </resultMap>
+
+    <insert id='insertUser' parameterType='User' useGeneratedKeys='true' keyProperty='id'>
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:14]: id="insert"

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 34]
```diff
@@ -0,0 +1,44 @@
...
+        INSERT INTO "user" (login, password, email, role_id)
+        VALUES(#{login}, #{password}, #{email}, #{role})
+    </insert>
+
+    <select id='getUserById' parameterType='long' resultType='User'>
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:19]: id="findById"

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 35]
```diff
@@ -0,0 +1,44 @@
...
+            id as id,
+            login as login,
+            password,
+            email as email,
+            role_id as role
+        FROM "user"
+        WHERE id = #{id}
+    </select>
+
+    <update id='updateUser' parameterType='User'>
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:30]: id="udpate"

File: src/main/resources/db/mybatis/UserMapper.xml [hunk 36]
```diff
@@ -0,0 +1,44 @@
...
+        UPDATE "user"
+        SET
+            login = #{login},
+            password = #{password},
+            email = #{email},
+            role_id = #{role}
+        WHERE id = #{id}
+    </update>
+
+    <delete id='deleteUser' parameterType='long'>
```
Comment (andrei-kuzniatsou) [inline src/main/resources/db/mybatis/UserMapper.xml:40]: id="delete"

File: src/main/java/com/epam/university/commission/mappers/IUserMapper.java (hunk 32 above)
Comment (SergeyPrudnikov-dev) [reply src/main/java/com/epam/university/commission/mappers/IUserMapper.java:6]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 33 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml:14]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 34 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml:19]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 35 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml:30]: Fixed.

File: src/main/resources/db/mybatis/UserMapper.xml (hunk 36 above)
Comment (SergeyPrudnikov-dev) [reply src/main/resources/db/mybatis/UserMapper.xml:40]: Fixed.

**Review by andrei-kuzniatsou — APPROVED** [review]

**Pull request merged**